pallets/funding/src/tests/3_auction.rs

```diff
@@ -1,1300 +1,1699 @@
+
+			// bid that fills 80% of the first bucket
+			let bid_40_percent = MockInstantiator::generate_bids_from_total_ct_percent(
+				project_metadata.clone(),
+				40u8,
+				vec![100],
+				vec![BIDDER_1],
+				vec![8u8],
+			);
+
+			// Note: 5% of total CTs is one bucket, i.e 10% of the auction allocation
+			// This bid fills last 20% of the first bucket,
+			// and gets split into 3 more bids of 2 more full and one partially full buckets.
+			// 10% + 5% + 5% + 3% = 23%
+			let bid_23_percent = MockInstantiator::generate_bids_from_total_ct_percent(
+				project_metadata.clone(),
+				23u8,
+				vec![100],
+				vec![BIDDER_2],
+				vec![7u8],
+			);
+
+			let all_bids = vec![bid_40_percent[0].clone(), bid_23_percent[0].clone()];
+
+			let necessary_plmc = MockInstantiator::calculate_auction_plmc_charged_from_all_bids_made_or_with_bucket(
+				&all_bids,
+				project_metadata.clone(),
+				None,
+			);
+			let ed_plmc = necessary_plmc.accounts().existential_deposits();
+			let necessary_usdt =
+				MockInstantiator::calculate_auction_funding_asset_charged_from_all_bids_made_or_with_bucket(
+					&all_bids,
+					project_metadata.clone(),
+					None,
+				);
+			inst.mint_plmc_to(necessary_plmc.clone());
+			inst.mint_plmc_to(ed_plmc.clone());
+			inst.mint_foreign_asset_to(necessary_usdt.clone());
+
+			inst.bid_for_users(project_id, bid_40_percent.clone()).unwrap();
+			let stored_bids = inst.execute(|| Bids::<TestRuntime>::iter_prefix_values((project_id,)).collect_vec());
+			assert_eq!(stored_bids.len(), 1);
+
+			inst.bid_for_users(project_id, bid_23_percent.clone()).unwrap();
+			let mut stored_bids = inst.execute(|| Bids::<TestRuntime>::iter_prefix_values((project_id,)).collect_vec());
+			stored_bids.sort_by(|a, b| a.id.cmp(&b.id));
+			// 40% + 10% + 5% + 5% + 3% = 5 total bids
+			assert_eq!(stored_bids.len(), 5);
+
+			assert_eq!(stored_bids[1].original_ct_usd_price, PriceOf::<TestRuntime>::from_float(1.0));
+			assert_eq!(
+				stored_bids[1].original_ct_amount,
+				Percent::from_percent(10) * project_metadata.total_allocation_size
+			);
+			assert_eq!(stored_bids[2].original_ct_usd_price, PriceOf::<TestRuntime>::from_rational(11, 10));
+			assert_eq!(
+				stored_bids[2].original_ct_amount,
+				Percent::from_percent(5) * project_metadata.total_allocation_size
+			);
+
+			assert_eq!(stored_bids[3].original_ct_usd_price, PriceOf::<TestRuntime>::from_float(1.2));
+			assert_eq!(
+				stored_bids[3].original_ct_amount,
+				Percent::from_percent(5) * project_metadata.total_allocation_size
+			);
+
+			assert_eq!(stored_bids[4].original_ct_usd_price, PriceOf::<TestRuntime>::from_float(1.3));
+			assert_eq!(
+				stored_bids[4].original_ct_amount,
+				Percent::from_percent(3) * project_metadata.total_allocation_size
+			);
+			let current_bucket = inst.execute(|| Buckets::<TestRuntime>::get(project_id)).unwrap();
+			assert_eq!(current_bucket.current_price, PriceOf::<TestRuntime>::from_float(1.3));
```

Comment:
Nit: Maybe also check amount of tokens left in bucket (and other bucket fields?).